lib/order.rb

```diff
@@ -0,0 +1,91 @@
+require_relative 'customer'
+
+class Order
+	attr_reader :id
+	attr_accessor :products, :customer, :fulfillment_status
+
+	def initialize(id, products, customer, fulfillment_status = :pending)
+		@id = id
+		@products = products
+		@customer = customer
+		self.fulfillment_status = fulfillment_status
+	end
+
+	# Raises argument errors if status is not recognized
+	def fulfillment_status=(value)
+		potential_status = [:pending, :paid, :processing, :shipped, :complete]
+		if potential_status.include? value
+			@fulfillment_status=(value)
+		else
+			raise ArgumentError,"Status is bogus"
+		end
+	end 
+
+	# Calculates the total cost of order, plus tax
+	def total
+		if @products == {}
```

Comment:
A slightly clearer way to do this is:

```suggestion
		if @products.empty?
```